### PR TITLE
rpc, docs: Add note for commands that supports only legacy wallets

### PR DESCRIPTION
**Pull request by yusufsahinhamza**
Refs #25363, apparently issue is not updated since over a month, so i decided to put the same `importaddress` note in #25368 to other rpc commands that needs this note.

Note is added for following commands:

- `importprivkey`
- `importpubkey`
- `importwallet`
- `dumpprivkey`
- `dumpwallet`
- `importmulti`
- `addmultisigaddress`
- `sethdseed`

### CODE REVIEW

**Review by jonatack — COMMENTED** [review]
This note you have copied over from #25368 does not seem relevant to these RPCs. In some cases, the error message may need to be updated as well.

**Comment by yusufsahinhamza** [comment]
@jonatack 

Looks like no need of this note for `getnewaddress`.

According to this release note:

https://github.com/bitcoin/bitcoin/blob/194f6dc43ccc330a8a4607be3a2b8935490d6db0/doc/release-notes/release-notes-0.21.0.md?plain=1#L412-L422

We can simply add **"Note: This command is only compatible with legacy wallets."** note to these RPCs, but for example where there is alternative command to use for descriptor wallets; e.g. release note says:

https://github.com/bitcoin/bitcoin/blob/194f6dc43ccc330a8a4607be3a2b8935490d6db0/doc/release-notes/release-notes-0.21.0.md?plain=1#L405-L406

In this case for `importmulti` we can add additional note **"Use "importdescriptors" with addr(X)" for descriptor wallets."** after the first note to point out what to use as an alternative.

**Review by Onyeali — APPROVED** [review]

**Comment by luke-jr** [comment]
>getnewaddress

?!?!?!

**Comment by yusufsahinhamza** [comment]
> >getnewaddress
> 
> 
> 
> ?!?!?!

Didn't do any changes as i'm waiting for an answer about my last message, or let's say suggestion to what to do.

File: src/wallet/rpc/backup.cpp [hunk 1]
```diff
@@ -1258,7 +1263,8 @@ RPCHelpMan importmulti()
             "may report that the imported keys, addresses or scripts exist but related transactions are still missing.\n"
             "The rescan parameter can be set to false if the key was never used to create transactions. If it is set to false,\n"
             "but the key was used to create transactions, rescanwallet needs to be called with the appropriate block range.\n"
-            "Note: Use \"getwalletinfo\" to query the scanning progress.\n",
+            "Note: Use \"getwalletinfo\" to query the scanning progress.\n"
+            "Note: This command is only compatible with legacy wallets. Use \"importdescriptors\" with \"addr(X)\" for descriptor wallets.\n",
```
Comment (achow101) [inline src/wallet/rpc/backup.cpp:1267]: I don't think it's useful to mention `addr()` here as `importmulti` allows the import of a ton of different things, including descriptors. Mentioning a specific descriptor type, which only applies to one of the things that `importmulti` can deal with, doesn't seem to make sense to me.

Comment (yusufsahinhamza) [reply src/wallet/rpc/backup.cpp:1267]: @achow101 I agree, I'd rather like it as **"Use "importdescriptors" for descriptor wallets."**, mentioning a specific parameter doesn't seem necessary. People would get help by using only **"importdescriptors"** anyway.

This line can also be changed:
https://github.com/bitcoin/bitcoin/blob/1abbae65eb9c99df5d8941008068d83ad99bf117/src/wallet/rpc/backup.cpp#L213

**Comment by tusher456** [comment]
32sMEMSwnLLcuxzHptTDksk9auSPoccWiJ

**Comment by yusufsahinhamza** [comment]
775ba9ccea0adcae349787647e8aa08844ff390d Conflicts are resolved and rebased.

**Review by theStack — COMMENTED** [review]
Concept ACK

For the `importprivkey` and `importpubkey` RPC notes, it could make sense to suggest using `importdescriptor` with `combo(X)` for descriptor wallets?

**Comment by yusufsahinhamza** [comment]
> For the `importprivkey` and `importpubkey` RPC notes, it could make sense to suggest using `importdescriptor` with `combo(X)` for descriptor wallets?

@theStack It seems to make sense for me. I can rebase and push changes if no other suggestion about that.

**Comment by achow101** [comment]
ACK 9141e4395a5f923059ad61ac6ba42a1a89e92cb0

**Comment by luke-jr** [comment]
@achow101 Why was this merged?

I'm not sure suggesting `combo()` is a good idea. Isn't combo only for having a migration path, not something we want people to actually use?

**Comment by achow101** [comment]
While I don't think we should necessarily encourage people to use `combo()`, I don't think it's bad to mention it either. in the case of `importpubkey` and `importprivkey`, `combo()` with `importdescriptors` does replicate their behavior so it isn't incorrect. I don't really care either way whether it's mentioned or not.

**Pull request merged**